src/main/java/newbank/server/commands/DepositCommand.java

```diff
@@ -0,0 +1,54 @@
+package newbank.server.commands;
+
+import org.javamoney.moneta.Money;
+
+import newbank.server.CustomerID;
+import newbank.server.NewBank;
+import newbank.server.exceptions.RequestNotAllowedException;
+
+public class DepositCommand extends Command {
+  private final NewBank bank;
+  private final String[] tokens;
+  private final CustomerID customer;
+
+  public DepositCommand(final NewBank bank, final String[] tokens, final CustomerID customer) {
+    this.bank = bank;
+    this.tokens = tokens;
+    this.customer = customer;
+  }
+
+  protected String getSyntax() {
+    return "DEPOSIT <account_name> <amount>";
+  }
+
+  private boolean isAccountValid(final String accountName) {
+    return bank.hasAccount(customer, accountName);
+  }
+
+  @Override
+  public String execute() {
+    if ((tokens.length >= 2 && tokens[1].equalsIgnoreCase("HELP"))) {
+      return String.format("SUCCESS: Usage: %s", getSyntax());
+    }
+
+    try {
+      checkLoggedIn(customer);
+    } catch (RequestNotAllowedException ex) {
+      return ex.getMessage();
+    }
+
+    if (tokens.length != 3) {
+      return String.format("FAIL: Usage: %s", getSyntax());
+    }
+
+    final String accountName = tokens[1];
+
+    if (!isAccountValid(accountName)) {
+      return String.format("FAIL: Account [%s] does not exist.", accountName);
+    }
+
+    bank.depositMoney(customer, accountName, Money.of(Double.valueOf(tokens[2]), "GBP"));
```

Comment:
There's clearly some rework here that's not ideal. We iterate through the customer's accounts to check there's one with the name `accountName`; if there is, we iterate again to (re!)find that account and deposit in it.
Could we not instead either:
- Get an `Optional<Account>` account on line 46. If it's not `Optional#present`, return the error message. If it is present, pass a reference to the `Account` to `NewBack#depositMoney`? OR...
- Just call `NewBank#depositMoney`, and have it return a boolean: if it returns `false`, no such account exists; if it returns `true` the deposit succeeded?

Reply:
Refactored.

Kindest Regards,
Jules